apps/stripe/src/app/api/stripe/webhook/stripe-object-handlers/stripe-payment-intent-handler.ts

```diff
@@ -0,0 +1,131 @@
+import { err, ok, Result } from "neverthrow";
+import Stripe from "stripe";
+
+import { SaleorMoney } from "@/modules/saleor/saleor-money";
+import { createDateFromStripeEvent } from "@/modules/stripe/stripe-event-date";
+import { StripePaymentIntentId } from "@/modules/stripe/stripe-payment-intent-id";
+import { createStripePaymentIntentStatus } from "@/modules/stripe/stripe-payment-intent-status";
+import {
+  AuthorizationFailureResult,
+  ChargeFailureResult,
+} from "@/modules/transaction-result/failure-result";
+import { mapPaymentIntentStatusToTransactionResult } from "@/modules/transaction-result/map-payment-intent-status-to-transaction-result";
+import { RecordedTransaction } from "@/modules/transactions-recording/domain/recorded-transaction";
+
+import { TransactionEventReportVariablesResolver } from "../transaction-event-report-variables-resolver";
+
+export type SupportedEvents =
+  | Stripe.PaymentIntentSucceededEvent
+  | Stripe.PaymentIntentProcessingEvent
+  | Stripe.PaymentIntentRequiresActionEvent
+  | Stripe.PaymentIntentAmountCapturableUpdatedEvent
+  | Stripe.PaymentIntentPaymentFailedEvent;
+
+export class StripePaymentIntentHandler {
```

Comment:
if this is suppossed to work only for webhook, maybe add "event" to name so we dont try to use it elsewhere